apps/emqx_connector_aggregator/src/emqx_connector_aggreg_delivery.erl

```diff
@@ -0,0 +1,188 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2022-2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+%% This module takes aggregated records from a buffer and delivers them to a blob storage
+%% backend, wrapped in a configurable container (though currently there's only CSV).
+-module(emqx_connector_aggreg_delivery).
+
+-include_lib("snabbkaffe/include/trace.hrl").
+-include("emqx_connector_aggregator.hrl").
+
+-export([start_link/3]).
+
+%% Internal exports
+-export([
+    init/4,
+    loop/3
+]).
+
+%% Sys
+-export([
```

Comment:
I initially thought that this way it might look simpler in the end, when compared to `gen_statem` with its overhead for basically an "interruptible task". But apparently it does not, so that's probably a good idea to adopt them.